components/nexus-base/src/main/java/org/sonatype/nexus/internal/app/ApplicationDirectoriesImpl.java

```diff
@@ -47,12 +47,13 @@
 
   @Inject
   public ApplicationDirectoriesImpl(@Named("${karaf.base}") final File installDir,
-                                    @Named("${karaf.data}") final File workDir)
+                                    @Named("${karaf.data}") final File workDir,
+                                    @Named("${karaf.etc}") final File configDir)
   {
     this.installDir = resolve(installDir, false);
     log.debug("Install dir: {}", this.installDir);
 
-    this.configDir = resolve(new File(installDir, "etc"), false);
+    this.configDir = resolve(configDir, false);
```

Comment:
The original line is meant to be that way - we've split up the configuration area into different subsystems to make it easier to manage, so Karaf-specific configuration files (under `karaf.etc` which now defaults to `$install-dir/etc/karaf`) are separate from Jetty configuration files (now under `jetty.etc` which defaults to `$install-dir/etc/jetty`) rather than have everything in the same directory.

Right now these are all located relative to the install directory for the sake of simplicity/consistency, ease of upgrade, and the ability to quickly switch between installations when upgrading (because certain configuration elements may change between releases it's harder to manage upgrades when the configuration is totally disconnected from the installation base)